File: android-kotlin/src/main/java/com/yourcompany/affiliatesdk/AffiliateSDK.kt
package com.yourcompany.affiliatesdk

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.net.ConnectivityManager
import android.net.NetworkCapabilities
import android.os.Build
import android.provider.Settings
import android.util.DisplayMetrics
import android.util.Log
import androidx.lifecycle.DefaultLifecycleObserver
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.ProcessLifecycleOwner
import kotlinx.coroutines.*
import okhttp3.*
import org.json.JSONObject
import java.io.IOException
import java.util.*
import java.util.concurrent.TimeUnit

/**
 * Configuration data class for AffiliateSDK
 */
data class AffiliateSDKConfig(
    val affiliateCode: String,
    val appCode: String,
    val baseUrl: String = "https://affiliate.33rd.pro/api/tracker.php",
    val debug: Boolean = false
)

/**
 * Purchase data class for tracking purchases
 */
data class PurchaseData(
    val amount: Double,
    val currency: String = "USD",
    val productId: String,
    val transactionId: String? = null,
    val additionalData: Map<String, Any>? = null
) {
    fun toMap(): Map<String, Any> {
        val map = mutableMapOf<String, Any>(
            "amount" to amount,
            "currency" to currency,
            "product_id" to productId
        )
        
        transactionId?.let { map["transaction_id"] = it }
        additionalData?.let { map.putAll(it) }
        
        return map
    }
}

/**
 * Main AffiliateSDK class for Android
 */
class AffiliateSDK private constructor() : DefaultLifecycleObserver {
    
    companion object {
        @Volatile
        private var INSTANCE: AffiliateSDK? = null
        private const val TAG = "AffiliateSDK"
        private const val PREFS_NAME = "affiliate_sdk_prefs"
        private const val SESSION_TIMEOUT_MS = 30 * 60 * 1000L // 30 minutes
        
        /**
         * Configure the SDK with required parameters
         */
        fun configure(
            context: Context,
            affiliateCode: String,
            appCode: String,
            baseUrl: String = "https://affiliate.33rd.pro/api/tracker.php",
            debug: Boolean = false
        ): AffiliateSDK {
            return getInstance().apply {
                this.context = context.applicationContext
                this.config = AffiliateSDKConfig(affiliateCode, appCode, baseUrl, debug)
            }
        }
        
        /**
         * Get singleton instance
         */
        fun getInstance(): AffiliateSDK {
            return INSTANCE ?: synchronized(this) {
                INSTANCE ?: AffiliateSDK().also { INSTANCE = it }
            }
        }
        
        /**
         * Initialize the SDK (convenience method)
         */
        suspend fun initialize() {
            getInstance().initialize()
        }
        
        /**
         * Track event (convenience method)
         */
        suspend fun trackEvent(eventName: String, parameters: Map<String, Any> = emptyMap()) {
            getInstance().trackEvent(eventName, parameters)
        }
        
        /**
         * Track screen view (convenience method)
         */
        suspend fun trackScreenView(screenName: String, parameters: Map<String, Any> = emptyMap()) {
            getInstance().trackScreenView(screenName, parameters)
        }
        
        /**
         * Track purchase (convenience method)
         */
        suspend fun trackPurchase(purchaseData: PurchaseData) {
            getInstance().trackPurchase(purchaseData)
        }
        
        /**
         * Track button click (convenience method)
         */
        suspend fun trackButtonClick(buttonId: String, parameters: Map<String, Any> = emptyMap()) {
            getInstance().trackButtonClick(buttonId, parameters)
        }
        
        /**
         * Track form submit (convenience method)
         */
        suspend fun trackFormSubmit(formName: String, parameters: Map<String, Any> = emptyMap()) {
            getInstance().trackFormSubmit(formName, parameters)
        }
    }
    
    private lateinit var context: Context
    private lateinit var config: AffiliateSDKConfig
    private lateinit var prefs: SharedPreferences
    private lateinit var okHttpClient: OkHttpClient
    
    private var sessionId: String? = null
    private var sessionStartTime: Long = 0
    private var lastEventTime: Long = 0
    private var isInitialized = false
    
    private var deviceInfo: Map<String, Any> = emptyMap()
    private val eventQueue = mutableListOf<Map<String, Any>>()
    
    private val platform = "android"
    private val scope = CoroutineScope(Dispatchers.IO + SupervisorJob())
    
    /**
     * Initialize the SDK
     */
    suspend fun initialize() {
        if (!::context.isInitialized || !::config.isInitialized) {
            logError("SDK not configured. Call configure() first.")
            return
        }
        
        if (isInitialized) {
            log("SDK already initialized")
            return
        }
        
        try {
            // Initialize preferences and HTTP client
            prefs = context.getSharedPreferences("${PREFS_NAME}_${config.affiliateCode}", Context.MODE_PRIVATE)
            okHttpClient = OkHttpClient.Builder()
                .connectTimeout(10, TimeUnit.SECONDS)
                .readTimeout(10, TimeUnit.SECONDS)
                .build()
            
            // Generate or restore session ID
            sessionId = getOrCreateSessionId()
            sessionStartTime = System.currentTimeMillis()
            lastEventTime = System.currentTimeMillis()
            
            // Collect device information
            collectDeviceInfo()
            
            // Process any queued events
            processEventQueue()
            
            // Track app launch
            val launchData = deviceInfo.toMutableMap().apply {
                put("session_id", sessionId ?: "")
                put("platform", platform)
            }
            
            trackEvent("app_launch", launchData)
            
            // Setup lifecycle observer
            if (context is Application) {
                ProcessLifecycleOwner.get().lifecycle.addObserver(this)
            }
            
            isInitialized = true
            log("SDK initialized successfully")
            
        } catch (e: Exception) {
            logError("Failed to initialize SDK: ${e.message}")
        }
    }
    
    /**
     * Track a custom event
     */
    suspend fun trackEvent(eventName: String, parameters: Map<String, Any> = emptyMap()) {
        if (!::config.isInitialized) {
            logError("SDK not configured")
            return
        }
        
        val eventData = mutableMapOf<String, Any>(
            "affiliate_code" to config.affiliateCode,
            "app_code" to config.appCode,
            "event" to eventName,
            "timestamp" to System.currentTimeMillis(),
            "session_id" to (sessionId ?: ""),
            "platform" to platform
        ).apply {
            putAll(parameters)
        }
        
        sendEvent(eventData)
        lastEventTime = System.currentTimeMillis()
    }
    
    /**
     * Track screen view
     */
    suspend fun trackScreenView(screenName: String, parameters: Map<String, Any> = emptyMap()) {
        val params = parameters.toMutableMap().apply {
            put("screen_name", screenName)
        }
        trackEvent("screen_view", params)
    }
    
    /**
     * Track purchase event
     */
    suspend fun trackPurchase(purchaseData: PurchaseData) {
        trackEvent("purchase", purchaseData.toMap())
    }
    
    /**
     * Track button click
     */
    suspend fun trackButtonClick(buttonId: String, parameters: Map<String, Any> = emptyMap()) {
        val params = parameters.toMutableMap().apply {
            put("button_id", buttonId)
        }
        trackEvent("button_click", params)
    }
    
    /**
     * Track form submission
     */
    suspend fun trackFormSubmit(formName: String, parameters: Map<String, Any> = emptyMap()) {
        val params = parameters.toMutableMap().apply {
            put("form_name", formName)
        }
        trackEvent("form_submit", params)
    }
    
    /**
     * Set user properties
     */
    fun setUserProperties(properties: Map<String, Any>) {
        try {
            val json = JSONObject(properties).toString()
            prefs.edit().putString("user_properties", json).apply()
        } catch (e: Exception) {
            logError("Failed to set user properties: ${e.message}")
        }
    }
    
    /**
     * Get user properties
     */
    fun getUserProperties(): Map<String, Any> {
        return try {
            val json = prefs.getString("user_properties", "{}")
            val jsonObject = JSONObject(json ?: "{}")
            val map = mutableMapOf<String, Any>()
            
            jsonObject.keys().forEach { key ->
                map[key] = jsonObject.get(key)
            }
            
            map
        } catch (e: Exception) {
            logError("Failed to get user properties: ${e.message}")
            emptyMap()
        }
    }
    
    /**
     * Track session end
     */
    suspend fun trackSessionEnd() {
        if (sessionStartTime == 0L) return
        
        val sessionDuration = System.currentTimeMillis() - sessionStartTime
        trackEvent("session_end", mapOf(
            "duration" to sessionDuration,
            "session_id" to (sessionId ?: "")
        ))
    }
    
    // Lifecycle callbacks
    override fun onStop(owner: LifecycleOwner) {
        scope.launch {
            trackSessionEnd()
        }
    }
    
    override fun onStart(owner: LifecycleOwner) {
        scope.launch {
            updateSessionIfNeeded()
        }
    }
    
    // Private methods
    
    private fun collectDeviceInfo() {
        try {
            val packageManager = context.packageManager
            val packageInfo = packageManager.getPackageInfo(context.packageName, 0)
            val displayMetrics = context.resources.displayMetrics
            
            val info = mutableMapOf<String, Any>(
                "device_id" to getDeviceId(),
                "device_model" to Build.MODEL,
                "device_brand" to Build.BRAND,
                "device_manufacturer" to Build.MANUFACTURER,
                "os_name" to "android",
                "os_version" to Build.VERSION.RELEASE,
                "sdk_int" to Build.VERSION.SDK_INT,
                "app_version" to packageInfo.versionName,
                "app_build" to if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
                    packageInfo.longVersionCode.toString()
                } else {
                    @Suppress("DEPRECATION")
                    packageInfo.versionCode.toString()
                },
                "package_name" to context.packageName,
                "screen_width" to displayMetrics.widthPixels,
                "screen_height" to displayMetrics.heightPixels,
                "screen_density" to displayMetrics.density,
                "locale" to Locale.getDefault().toString(),
                "timezone" to TimeZone.getDefault().id
            )
            
            deviceInfo = info
        } catch (e: Exception) {
            logError("Failed to collect device info: ${e.message}")
            deviceInfo = mapOf(
                "os_name" to "android",
                "platform" to platform
            )
        }
    }
    
    private fun getDeviceId(): String {
        return try {
            Settings.Secure.getString(context.contentResolver, Settings.Secure.ANDROID_ID)
                ?: "unknown_device"
        } catch (e: Exception) {
            "unknown_device"
        }
    }
    
    private fun getOrCreateSessionId(): String {
        val existingSessionId = prefs.getString("session_id", null)
        
        if (existingSessionId != null) {
            return existingSessionId
        }
        
        val newSessionId = "sess_${System.currentTimeMillis()}_${generateRandomString(9)}"
        prefs.edit().putString("session_id", newSessionId).apply()
        return newSessionId
    }
    
    private suspend fun sendEvent(eventData: Map<String, Any>) {
        try {
            val urlBuilder = HttpUrl.parse(config.baseUrl)?.newBuilder()
                ?: throw IOException("Invalid base URL")
            
            eventData.forEach { (key, value) ->
                urlBuilder.addQueryParameter(key, value.toString())
            }
            
            val request = Request.Builder()
                .url(urlBuilder.build())
                .get()
                .build()
            
            withContext(Dispatchers.IO) {
                okHttpClient.newCall(request).execute().use { response ->
                    if (response.isSuccessful) {
                        log("Event sent successfully: ${eventData["event"]}")
                    } else {
                        throw IOException("HTTP ${response.code}: ${response.message}")
                    }
                }
            }
            
        } catch (e: Exception) {
            logError("Failed to send event: ${e.message}")
            
            // Queue event for retry
            val queuedEvent = eventData.toMutableMap().apply {
                put("retry_count", 0)
                put("queued_at", System.currentTimeMillis())
            }
            eventQueue.add(queuedEvent)
        }
    }
    
    private suspend fun processEventQueue() {
        if (eventQueue.isEmpty()) return
        
        val eventsToProcess = eventQueue.toList()
        eventQueue.clear()
        
        eventsToProcess.forEach { event ->
            val retryCount = event["retry_count"] as? Int ?: 0
            
            if (retryCount < 3) {
                try {
                    val eventToSend = event.toMutableMap().apply {
                        remove("retry_count")
                        remove("queued_at")
                    }
                    sendEvent(eventToSend)
                } catch (e: Exception) {
                    // Re-queue with incremented retry count
                    val requeuedEvent = event.toMutableMap().apply {
                        put("retry_count", retryCount + 1)
                    }
                    eventQueue.add(requeuedEvent)
                }
            }
        }
    }
    
    private suspend fun updateSessionIfNeeded() {
        if (lastEventTime == 0L) return
        
        val timeSinceLastEvent = System.currentTimeMillis() - lastEventTime
        
        if (timeSinceLastEvent > SESSION_TIMEOUT_MS) {
            // Start new session
            sessionId = getOrCreateSessionId()
            sessionStartTime = System.currentTimeMillis()
            
            trackEvent("app_launch", mapOf(
                "session_id" to (sessionId ?: ""),
                "returning_user" to true
            ))
        }
    }
    
    private fun generateRandomString(length: Int): String {
        val chars = "abcdefghijklmnopqrstuvwxyz0123456789"
        return (1..length)
            .map { chars.random() }
            .joinToString("")
    }
    
    private fun log(message: String) {
        if (::config.isInitialized && config.debug) {
            Log.d(TAG, message)
        }
    }
    
    private fun logError(message: String) {
        if (::config.isInitialized && config.debug) {
            Log.e(TAG, message)
        }
    }
}

File: flutter/lib/affiliate_sdk.dart
library affiliate_sdk;

import 'dart:async';
import 'dart:convert';
import 'dart:io';
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:http/http.dart' as http;
import 'package:shared_preferences/shared_preferences.dart';
import 'package:device_info_plus/device_info_plus.dart';
import 'package:package_info_plus/package_info_plus.dart';
import 'package:connectivity_plus/connectivity_plus.dart';

/// Configuration class for AffiliateSDK
class AffiliateSDKConfig {
  final String affiliateCode;
  final String appCode;
  final String? baseUrl;
  final bool debug;

  const AffiliateSDKConfig({
    required this.affiliateCode,
    required this.appCode,
    this.baseUrl,
    this.debug = false,
  });
}

/// Purchase data for tracking purchases
class PurchaseData {
  final double amount;
  final String currency;
  final String productId;
  final String? transactionId;
  final Map<String, dynamic>? additionalData;

  const PurchaseData({
    required this.amount,
    this.currency = 'USD',
    required this.productId,
    this.transactionId,
    this.additionalData,
  });

  Map<String, dynamic> toMap() {
    return {
      'amount': amount,
      'currency': currency,
      'product_id': productId,
      if (transactionId != null) 'transaction_id': transactionId,
      if (additionalData != null) ...additionalData!,
    };
  }
}

/// Main AffiliateSDK class
class AffiliateSDK {
  static AffiliateSDK? _instance;
  late AffiliateSDKConfig _config;
  
  String? _sessionId;
  DateTime? _sessionStartTime;
  DateTime? _lastEventTime;
  bool _isInitialized = false;
  
  Map<String, dynamic> _deviceInfo = {};
  List<Map<String, dynamic>> _eventQueue = [];
  
  final String _platform = 'flutter';
  late String _baseUrl;
  
  // Private constructor
  AffiliateSDK._();
  
  /// Get singleton instance
  static AffiliateSDK get instance {
    _instance ??= AffiliateSDK._();
    return _instance!;
  }
  
  /// Factory constructor
  factory AffiliateSDK({
    required String affiliateCode,
    required String appCode,
    String? baseUrl,
    bool debug = false,
  }) {
    final instance = AffiliateSDK.instance;
    instance._config = AffiliateSDKConfig(
      affiliateCode: affiliateCode,
      appCode: appCode,
      baseUrl: baseUrl,
      debug: debug,
    );
    instance._baseUrl = baseUrl ?? 'https://affiliate.33rd.pro/api/tracker.php';
    return instance;
  }

  /// Initialize the SDK
  Future<void> initialize() async {
    try {
      if (_isInitialized) {
        _log('SDK already initialized');
        return;
      }

      // Generate or restore session ID
      _sessionId = await _getOrCreateSessionId();
      _sessionStartTime = DateTime.now();
      _lastEventTime = DateTime.now();

      // Collect device information
      await _collectDeviceInfo();

      // Process any queued events
      await _processEventQueue();

      // Track app launch
      await trackEvent('app_launch', {
        'session_id': _sessionId,
        'platform': _platform,
        ..._deviceInfo,
      });

      _isInitialized = true;
      _log('SDK initialized successfully');

      // Setup app lifecycle listener
      _setupAppLifecycleListener();

    } catch (error) {
      _logError('Failed to initialize SDK: $error');
    }
  }

  /// Track a custom event
  Future<void> trackEvent(String eventName, [Map<String, dynamic>? parameters]) async {
    try {
      final eventData = {
        'affiliate_code': _config.affiliateCode,
        'app_code': _config.appCode,
        'event': eventName,
        'timestamp': DateTime.now().millisecondsSinceEpoch,
        'session_id': _sessionId,
        'platform': _platform,
        if (parameters != null) ...parameters,
      };

      await _sendEvent(eventData);
      _lastEventTime = DateTime.now();

    } catch (error) {
      _logError('Failed to track event: $error');
    }
  }

  /// Track screen view
  Future<void> trackScreenView(String screenName, [Map<String, dynamic>? parameters]) async {
    await trackEvent('screen_view', {
      'screen_name': screenName,
      if (parameters != null) ...parameters,
    });
  }

  /// Track purchase event
  Future<void> trackPurchase(PurchaseData purchaseData) async {
    await trackEvent('purchase', purchaseData.toMap());
  }

  /// Track button click
  Future<void> trackButtonClick(String buttonId, [Map<String, dynamic>? parameters]) async {
    await trackEvent('button_click', {
      'button_id': buttonId,
      if (parameters != null) ...parameters,
    });
  }

  /// Track form submission
  Future<void> trackFormSubmit(String formName, [Map<String, dynamic>? parameters]) async {
    await trackEvent('form_submit', {
      'form_name': formName,
      if (parameters != null) ...parameters,
    });
  }

  /// Set user properties
  Future<void> setUserProperties(Map<String, dynamic> properties) async {
    try {
      final prefs = await SharedPreferences.getInstance();
      await prefs.setString(
        'affiliate_sdk_user_props_${_config.affiliateCode}',
        jsonEncode(properties),
      );
    } catch (error) {
      _logError('Failed to set user properties: $error');
    }
  }

  /// Get user properties
  Future<Map<String, dynamic>> getUserProperties() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      final stored = prefs.getString('affiliate_sdk_user_props_${_config.affiliateCode}');
      return stored != null ? jsonDecode(stored) : {};
    } catch (error) {
      _logError('Failed to get user properties: $error');
      return {};
    }
  }

  /// Track session end
  Future<void> trackSessionEnd() async {
    if (_sessionStartTime == null) return;

    final sessionDuration = DateTime.now().difference(_sessionStartTime!).inMilliseconds;
    await trackEvent('session_end', {
      'duration': sessionDuration,
      'session_id': _sessionId,
    });
  }

  /// Get navigation observer for automatic screen tracking
  NavigatorObserver getNavigatorObserver() {
    return _AffiliateNavigatorObserver(this);
  }

  // Private methods

  Future<void> _collectDeviceInfo() async {
    try {
      final deviceInfo = DeviceInfoPlugin();
      final packageInfo = await PackageInfo.fromPlatform();
      
      Map<String, dynamic> info = {
        'app_version': packageInfo.version,
        'app_build': packageInfo.buildNumber,
        'package_name': packageInfo.packageName,
      };

      if (Platform.isAndroid) {
        final androidInfo = await deviceInfo.androidInfo;
        info.addAll({
          'device_id': androidInfo.id,
          'device_model': androidInfo.model,
          'device_brand': androidInfo.brand,
          'os_name': 'android',
          'os_version': androidInfo.version.release,
          'sdk_int': androidInfo.version.sdkInt,
        });
      } else if (Platform.isIOS) {
        final iosInfo = await deviceInfo.iosInfo;
        info.addAll({
          'device_id': iosInfo.identifierForVendor,
          'device_model': iosInfo.model,
          'device_brand': 'Apple',
          'os_name': 'ios',
          'os_version': iosInfo.systemVersion,
        });
      }

      // Get screen information
      final window = WidgetsBinding.instance.window;
      final size = window.physicalSize / window.devicePixelRatio;
      info.addAll({
        'screen_width': size.width.round(),
        'screen_height': size.height.round(),
        'pixel_ratio': window.devicePixelRatio,
      });

      _deviceInfo = info;
    } catch (error) {
      _logError('Failed to collect device info: $error');
      _deviceInfo = {
        'os_name': Platform.operatingSystem,
        'platform': _platform,
      };
    }
  }

  Future<String> _getOrCreateSessionId() async {
    try {
      final prefs = await SharedPreferences.getInstance();
      String? sessionId = prefs.getString('affiliate_sdk_session_${_config.affiliateCode}');
      
      if (sessionId == null) {
        sessionId = 'sess_${DateTime.now().millisecondsSinceEpoch}_${_generateRandomString(9)}';
        await prefs.setString('affiliate_sdk_session_${_config.affiliateCode}', sessionId);
      }
      
      return sessionId;
    } catch (error) {
      _logError('Failed to get/create session ID: $error');
      return 'sess_${DateTime.now().millisecondsSinceEpoch}_${_generateRandomString(9)}';
    }
  }

  Future<void> _sendEvent(Map<String, dynamic> eventData) async {
    try {
      final uri = Uri.parse(_baseUrl).replace(
        queryParameters: eventData.map((key, value) => MapEntry(key, value.toString())),
      );
      
      final response = await http.get(uri).timeout(const Duration(seconds: 10));

      if (response.statusCode == 200) {
        _log('Event sent successfully: ${eventData['event']}');
      } else {
        throw Exception('HTTP ${response.statusCode}: ${response.reasonPhrase}');
      }

    } catch (error) {
      _logError('Failed to send event: $error');
      
      // Queue event for retry
      _eventQueue.add({
        ...eventData,
        'retry_count': 0,
        'queued_at': DateTime.now().millisecondsSinceEpoch,
      });
    }
  }

  Future<void> _processEventQueue() async {
    if (_eventQueue.isEmpty) return;

    final eventsToProcess = List<Map<String, dynamic>>.from(_eventQueue);
    _eventQueue.clear();

    for (final event in eventsToProcess) {
      try {
        final retryCount = event['retry_count'] ?? 0;
        if (retryCount < 3) {
          event.remove('retry_count');
          event.remove('queued_at');
          await _sendEvent(event);
        }
      } catch (error) {
        // Re-queue with incremented retry count
        _eventQueue.add({
          ...event,
          'retry_count': (event['retry_count'] ?? 0) + 1,
        });
      }
    }
  }

  void _setupAppLifecycleListener() {
    SystemChannels.lifecycle.setMessageHandler((message) async {
      switch (message) {
        case 'AppLifecycleState.paused':
          await trackSessionEnd();
          break;
        case 'AppLifecycleState.resumed':
          await _updateSessionIfNeeded();
          break;
      }
      return null;
    });
  }

  Future<void> _updateSessionIfNeeded() async {
    if (_lastEventTime == null) return;
    
    final timeSinceLastEvent = DateTime.now().difference(_lastEventTime!);
    const sessionTimeout = Duration(minutes: 30);

    if (timeSinceLastEvent > sessionTimeout) {
      // Start new session
      _sessionId = await _getOrCreateSessionId();
      _sessionStartTime = DateTime.now();
      
      await trackEvent('app_launch', {
        'session_id': _sessionId,
        'returning_user': true,
      });
    }
  }

  String _generateRandomString(int length) {
    const chars = 'abcdefghijklmnopqrstuvwxyz0123456789';
    return List.generate(length, (index) => chars[(DateTime.now().millisecondsSinceEpoch + index) % chars.length]).join();
  }

  void _log(String message) {
    if (_config.debug) {
      debugPrint('[AffiliateSDK] $message');
    }
  }

  void _logError(String message) {
    if (_config.debug) {
      debugPrint('[AffiliateSDK Error] $message');
    }
  }
}

/// Navigator observer for automatic screen tracking
class _AffiliateNavigatorObserver extends NavigatorObserver {
  final AffiliateSDK _sdk;

  _AffiliateNavigatorObserver(this._sdk);

  @override
  void didPush(Route<dynamic> route, Route<dynamic>? previousRoute) {
    super.didPush(route, previousRoute);
    _trackScreenView(route);
  }

  @override
  void didReplace({Route<dynamic>? newRoute, Route<dynamic>? oldRoute}) {
    super.didReplace(newRoute: newRoute, oldRoute: oldRoute);
    if (newRoute != null) {
      _trackScreenView(newRoute);
    }
  }

  @override
  void didPop(Route<dynamic> route, Route<dynamic>? previousRoute) {
    super.didPop(route, previousRoute);
    if (previousRoute != null) {
      _trackScreenView(previousRoute);
    }
  }

  void _trackScreenView(Route<dynamic> route) {
    final routeName = route.settings.name;
    if (routeName != null && routeName.isNotEmpty) {
      _sdk.trackScreenView(routeName);
    }
  }
}

export 'affiliate_sdk.dart';